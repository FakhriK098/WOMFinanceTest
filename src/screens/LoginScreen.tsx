import React, { useCallback } from 'react';
import { StyleSheet, View, useColorScheme } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import AuthTemplate from '../components/templates/AuthTemplate';
import LoginForm from '../components/organisms/LoginForm';
import { AppDispatch, RootState } from '../store';
import { login } from '../store/slices/authSlice';
import Text from '../components/atoms/Text';
import { colors } from '../theme/colors';
import { spacing } from '../theme/spacing';
import { typography } from '../theme/typography';
import Spacer from '../components/atoms/Spacer';

const LoginScreen = () => {
  const dispatch = useDispatch<AppDispatch>();
  const { loading, error } = useSelector((s: RootState) => s.auth);
  const isDark = useColorScheme() === 'dark';
  const onSubmit = useCallback(
    (email: string, password: string) => {
      dispatch(login({ email, password }));
    },
    [dispatch],
  );
  return (
    <View
      style={[
        styles.wrapper,
        {
          backgroundColor: isDark
            ? colors.background.dark
            : colors.background.light,
        },
      ]}
    >
      <AuthTemplate title="Masuk">
        <Text
          size={typography.sm}
          color={isDark ? colors.text.mutedDark : colors.text.mutedLight}
        >
          Masuk untuk melanjutkan
        </Text>
        <Spacer size={spacing.lg} />
        <LoginForm onSubmit={onSubmit} loading={loading} error={error} />
      </AuthTemplate>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
  },
});

export default LoginScreen;
