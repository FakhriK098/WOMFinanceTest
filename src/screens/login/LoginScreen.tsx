import React, { useCallback } from 'react';
import { StyleSheet, View, useColorScheme } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { AuthTemplate, LoginForm, Text, Spacer } from '@components';
import { AppDispatch, RootState } from '@store';
import { login } from '@store/slices/authSlice';
import { colors } from '@theme/colors';
import { spacing } from '@theme/spacing';
import { typography } from '@theme/typography';

const LoginScreen = () => {
  const dispatch = useDispatch<AppDispatch>();
  const { loading, error } = useSelector((s: RootState) => s.auth);
  const isDark = useColorScheme() === 'dark';
  const onSubmit = useCallback(
    (email: string, password: string) => {
      dispatch(login({ email, password }));
    },
    [dispatch],
  );
  return (
    <View
      style={[
        styles.wrapper,
        {
          backgroundColor: isDark
            ? colors.background.dark
            : colors.background.light,
        },
      ]}
    >
      <AuthTemplate title="Masuk">
        <Text
          size={typography.sm}
          color={isDark ? colors.text.mutedDark : colors.text.mutedLight}
        >
          Masuk untuk melanjutkan
        </Text>
        <Spacer size={spacing.lg} />
        <LoginForm onSubmit={onSubmit} loading={loading} error={error} />
      </AuthTemplate>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
  },
});

export default LoginScreen;
