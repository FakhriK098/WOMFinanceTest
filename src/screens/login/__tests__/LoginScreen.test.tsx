import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
import * as RN from 'react-native';

import LoginScreen from '../LoginScreen';
jest.mock('@store', () => ({ store: {} }));
jest.mock('@store/slices/authSlice', () => ({ login: () => ({ type: 'login' }) }));

describe('LoginScreen', () => {
  const useSelectorSpy = useSelector as unknown as jest.Mock;
  const useDispatchSpy = useDispatch as unknown as jest.Mock;

  beforeEach(() => {
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    useDispatchSpy.mockReturnValue(jest.fn());
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders template and form (snapshot)', () => {
    useSelectorSpy.mockImplementation((sel: any) =>
      sel({ auth: { loading: false, error: null } }),
    );
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<LoginScreen />);
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const subtitle = tr!.root.find(
      (inst) => inst.props && inst.props.children === 'Masuk untuk melanjutkan',
    );
    expect(subtitle).toBeTruthy();
    act(() => tr!.unmount());
  });
});
