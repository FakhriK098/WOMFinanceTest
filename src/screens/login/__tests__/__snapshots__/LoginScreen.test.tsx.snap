// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders template and form (snapshot) 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
        },
        {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      style={
        {
          "borderRadius": 16,
          "gap": 8,
          "padding": 20,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#111827",
              "fontSize": 24,
              "fontWeight": "700",
            },
            undefined,
          ]
        }
      >
        Masuk
      </Text>
      <View
        style={
          {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 14,
              "fontWeight": "400",
            },
            undefined,
          ]
        }
      >
        Masuk untuk melanjutkan
      </Text>
      <View
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
      />
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#6B7280",
                  "fontSize": 14,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            Email
          </Text>
          <View
            style={
              {
                "height": 8,
                "width": 8,
              }
            }
          />
          <View
            style={
              [
                {
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "paddingHorizontal": 14,
                },
                {
                  "backgroundColor": "#1118270D",
                  "borderColor": "#E5E7EB",
                },
                false,
                false,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              keyboardType="email-address"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="nama@email.com"
              placeholderTextColor="#6B7280"
              style={
                [
                  {
                    "height": 48,
                  },
                  {
                    "color": "#111827",
                  },
                  undefined,
                ]
              }
              value=""
            />
          </View>
        </View>
        <View
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
        <View
          style={
            {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#6B7280",
                  "fontSize": 14,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            Password
          </Text>
          <View
            style={
              {
                "height": 8,
                "width": 8,
              }
            }
          />
          <View
            style={
              [
                {
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "paddingHorizontal": 14,
                },
                {
                  "backgroundColor": "#1118270D",
                  "borderColor": "#E5E7EB",
                },
                false,
                false,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              keyboardType="default"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="••••••"
              placeholderTextColor="#6B7280"
              secureTextEntry={true}
              style={
                [
                  {
                    "height": 48,
                  },
                  {
                    "color": "#111827",
                  },
                  undefined,
                ]
              }
              value=""
            />
          </View>
        </View>
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#4F46E5",
                "borderRadius": 12,
                "paddingVertical": 14,
              },
              {
                "opacity": 0.6,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "fontWeight": "700",
                  },
                  undefined,
                ]
              }
            >
              Masuk
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
